src/NHapi.Base/Model/Primitive/CommonTM.cs

```diff
@@ -24,609 +24,701 @@
+        {
+            set
+            {
+                try
+                {
+                    // When this function is called an offset is being created/updated
+                    // we should not omit displaying it in the return value from
+                    // the getValue() method
+                    omitOffsetFg = 'n';
+                    var offsetStr = Convert.ToString(value);
+                    if ((value >= 0 && offsetStr.Length > 4) || (value < 0 && offsetStr.Length > 5))
+                    {
+                        // The length of the GMT offset must be no greater than 5 characters (including the sign)
+                        var msg = "The length of the GMT offset for the TM datatype value does" +
+                                         " not conform to the allowable format [+/-ZZZZ]. Value: " + value;
+                        var e = new DataTypeException(msg);
+                        throw e;
+                    } // end if
+
+                    // obtain the absolute value of the input
+                    var absOffset = Math.Abs(value);
+
+                    // extract the hour data from the offset value.
+                    // first pre-append zeros so we have a 4 char offset value (without sign)
+                    offsetStr = DataTypeUtil.PreAppendZeroes(absOffset, 4);
+                    var hrOffsetInt = int.Parse(offsetStr.Substring(0, 2 - 0));
+
+                    // check to see if the hour value is valid
+                    if ((hrOffsetInt < 0) || (hrOffsetInt > 23))
+                    {
+                        var msg = "The GMT offset hour value of the TM datatype must be >=0 and <=23";
+                        var e = new DataTypeException(msg);
+                        throw e;
+                    } // end if
+
+                    // extract the minute data from the offset value.
+                    var minOffsetInt = int.Parse(offsetStr.Substring(2, 4 - 2));
+
+                    // check to see if the minute value is valid
+                    if ((minOffsetInt < 0) || (minOffsetInt > 59))
+                    {
+                        var msg = "The GMT offset minute value of the TM datatype must be >=0 and <=59";
+                        var e = new DataTypeException(msg);
+                        throw e;
+                    } // end if
+
+                    // The input value is valid, now store it in the offset field
+                    offSet = value;
+                }
+
+                // end try
+                catch (DataTypeException e)
+                {
+                    throw e;
+                }
+
+                // end catch
+                catch (Exception e)
+                {
+                    throw new DataTypeException("An unexpected exception occurred", e);
+                } // end catch
+            }
+        }
+
+        /// <summary> Returns the hour as an integer.</summary>
+        public virtual int Hour

```

Comment:
```csharp
public virtual int Hour { get; private set; }
```

same for Minute, Second, FractSecond, GTMOffset?

Reply:
done